common/persistence/cassandra/cassandraHistoryPersistence.go

```diff
@@ -130,63 +113,70 @@ func (h *cassandraHistoryV2Persistence) AppendHistoryNodes(
 	}
 
 	var err error
+	var treeRow *nosqlPlugin.HistoryTreeRow
 	if request.IsNewBranch {
-		ancs := []map[string]interface{}{}
-		for _, an := range branchInfo.Ancestors {
-			value := make(map[string]interface{})
-			value["end_node_id"] = *an.EndNodeID
-			value["branch_id"] = an.BranchID
-			ancs = append(ancs, value)
+		var ancestors []*shared.HistoryBranchRange
+		for _, anc := range branchInfo.Ancestors {
+			ancestors = append(ancestors, anc)
+		}
+		treeRow = &nosqlPlugin.HistoryTreeRow{
+			ShardID:                     request.ShardID,
+			TreeID:                      branchInfo.GetTreeID(),
+			BranchID:                    branchInfo.GetBranchID(),
+			Ancestors:                   ancestors,
+			CreateTimestampMilliseconds: p.UnixNanoToDBTimestamp(time.Now().UnixNano()),
+			Info:                        request.Info,
 		}
-
-		cqlNowTimestamp := p.UnixNanoToDBTimestamp(time.Now().UnixNano())
-		batch := h.session.NewBatch(gocql.LoggedBatch)
-		batch.Query(v2templateInsertTree,
-			branchInfo.TreeID, branchInfo.BranchID, ancs, cqlNowTimestamp, request.Info)
-		batch.Query(v2templateUpsertData,
-			branchInfo.TreeID, branchInfo.BranchID, request.NodeID, request.TransactionID, request.Events.Data, request.Events.Encoding)
-		err = h.session.ExecuteBatch(batch)
-	} else {
-		query := h.session.Query(v2templateUpsertData,
-			branchInfo.TreeID, branchInfo.BranchID, request.NodeID, request.TransactionID, request.Events.Data, request.Events.Encoding)
-		err = query.Exec()
 	}
+	nodeRow := &nosqlPlugin.HistoryNodeRow{
```

Comment:
These types will be different between sql and nosql?

Reply:
Yeah. There are some common pieces, we could move to some common pkg later if necessary. Right now they have to be different.

In SQL, we don't need `NextPageToken` in HistoryNodeRow. Because most SQL databse doesn't support paging token, so the sqlHistoryManager worked around it by using primary key.

But in Cassandra, we already done that. Ideally, we could do the same. Ideally we can do the samething, by using primary key, I guess. For the safty reason, I want to pass it here. It will add more flexibility to the database layer if they want to use paging token supported by underlying NoSQL DBs.

Also in SQL we decided later to use blob to represent the whole business logic including things like branchRanges for ancestors:
```	
HistoryTreeRow struct {
		ShardID      int
		TreeID       UUID
		BranchID     UUID
		Data         []byte
		DataEncoding string
	}
```
It turns out to be a better choice. Because using blob will help preventing non-compatibility changes to the interfaces, if we want to add some new fields.

However in Cassandra we already used specific fields just for `Ancestors` ,`CreateTime`, `Info`, etc. For backward compatibility, we have to keep passing them separately, if we don't want to waste CPU/memory for encoidng/decoding blobs. 

For new NoSQL DBs that we/community will implement later, ideally we should use blob as well. But I don't see a way to do that without avoiding compatibility issues yet. We could do that later if anyone has suggestion. 



So ideally